optimum/exporters/openvino/stateful.py

```diff
@@ -190,18 +190,93 @@ def ensure_stateful_is_available(warn=True):
     return True
 
 
-def ensure_export_task_support_stateful(task: str):
+_ENCODER_DECODER_TASKS_WITH_PAST = (
+    "automatic-speech-recognition",
+    "text2text-generation",
+)
+
+
+def ensure_export_task_support_stateful(task: str, is_encoder_decoder: bool = False):
```

Comment:
could there be any case where the task is "automatic-speech-recognition" or "text2text-generation" and the model is a decoder only model ? I'd imagine that the `task` argument should be enough here and that we can drop `is_encoder_decoder`

Reply:
it was copied from my old implementation that requires additional guidance for correct stateful seq2seq. there is also Vision2Seq models that also usually have 2 decoders, but not all - vision-encoder-decoder architecture uses gpt2 based model decoder that does not have cross attention, so it has only 1 decoder, but I do not see why model with single decoder can not be stateful if it works in the same way like in text-generation model task only getting image as input instead of text

Reply:
removed